src/components/ImageUploader/src/ImagePicker.vue

```diff
@@ -22,16 +35,118 @@
 
 <script>
 import PlusIcon from '@square/maker-icons/Plus';
+import ArrowIcon from '@square/maker-icons/ArrowUp';
+import {
+	isEqual, uniqWith, escapeRegExp, partition,
+} from 'lodash';
+
+const getEntryFile = (entry) => new Promise((resolve) => entry.file(resolve));
+const readEntries = (reader) => new Promise((resolve) => reader.readEntries(resolve));
+const getAllEntries = (reader, aggregated = []) => readEntries(reader).then((entries) => {
+	// eslint-disable-next-line unicorn/explicit-length-check
+	if (!entries.length) {
+		return aggregated;
+	}
+	return getAllEntries(reader, aggregated.concat(entries));
+});
+function traverseEntry(entry) {
+	if (entry) {
+		if (entry.isFile) {
+			return getEntryFile(entry);
+		}
+		if (entry.isDirectory) {
+			return getAllEntries(entry.createReader()).then(
```

Comment:
instead of using `then` here you can convert this function into an async function and use await